feat: Add Bright Data integration bundle - following PR #8579

Idanvilenski opened the pull request:
# 🚀 Bright Data Integration for Langflow

## Overview

This PR adds comprehensive **Bright Data integration** to Langflow, enabling users to perform enterprise-grade web scraping and data extraction directly within their AI workflows. Bright Data is the world's leading web data platform, trusted by Fortune 500 companies for reliable, ethical web data collection.

## ✨ Features Added

### 🧩 Three New Components

1. **BrightDataWebScraperComponent** - General web scraping with proxy support
2. **BrightDataSearchEngineComponent** - Search engine results extraction (SERP)
3. **BrightDataStructuredDataEnhancedComponent** - Structured data extraction with auto detect tool use

### 🛠️ Key Capabilities

- **🌐 Universal Web Scraping** - Extract data from any website without getting blocked
- **🔍 Search Engine Results** - Scrape Google, Bing, and other search engines
- **📊 Structured Data Extraction** - Clean, organized data using predefined schemas
- **🤖 Agent Integration** - Full tool mode support for AI agents
- **🛡️ Anti-Bot Protection** - Bypass CAPTCHAs and sophisticated bot detection
- **🌍 Global Proxy Network** - Access geo-restricted content from any location
- **⚡ Real-time Data** - Get fresh, up-to-date information instantly

## 🎯 Benefits for Langflow Users

- **No-code web scraping** capabilities in visual workflows
- **Enterprise-grade infrastructure** with 99.9% uptime
- **Ethical data collection** with compliance features
- **Easy integration** with existing agent workflows
- **Professional support** from Bright Data team

## 🔧 Technical Implementation

### Components Structure
```
src/backend/base/langflow/components/brightdata/
├── __init__.py                                    # Component registry
├── brightdata_search_engine.py                    # Search engine component
├── brightdata_structured_data.py                  # Structured data component
└── brightdata_web_scraper.py                     # Web scraper component
```

### Frontend Icons
```
src/frontend/src/icons/BrightData/
├── BrightData.jsx                                 # React icon component
├── BrightData.svg                                 # SVG icon
└── index.tsx                                      # TypeScript exports
```

### Component Features
- [x] **Tool Mode Support** - All components can be used as tools in agent workflows
- [x] **Type Safety** - Full TypeScript support with proper type annotations
- [x] **Error Handling** - Comprehensive error handling and user feedback
- [x] **Input Validation** - Robust input validation and sanitization
- [x] **Custom Icons** - Professional Bright Data branding
- [x] **Documentation** - Inline documentation and tooltips

## 🧪 Testing

- [x] **Lint checks pass** - `make lint` successful
- [x] **Components load** - All components appear in Langflow UI
- [x] **Tool mode works** - Successfully tested with agent components
- [x] **Error handling** - Proper error messages for invalid inputs
- [x] **Type safety** - All type annotations validated

## 📚 Usage Examples

### Basic Web Scraping
```python
# Scrape any website
url = "https://example.com"
content = web_scraper.scrape(url)
```

### Search Engine Results
```python
# Get search results
query = "AI workflows Langflow"
results = search_engine.search(query)
```

### Agent Integration
```yaml
Flow:
  - Chat Input: "Research AI trends"
  - Agent: OpenAI Tools Agent
  - Tools:
    - BrightDataSearchEngine
    - BrightDataWebScraper
  - Chat Output: Research summary
```

## 🔗 Dependencies

No new external dependencies added - uses existing Langflow infrastructure and standard Python libraries.

## 🏢 About Bright Data

- **Leading web data platform** serving 20,000+ customers
- **Fortune 500 trusted** by Microsoft, Samsung, Adobe
- **Ethical data collection** with full compliance
- **99.9% uptime** with enterprise SLA
- **24/7 support** for technical assistance

## 📞 Support & Maintenance

- **Active maintenance** - Bright Data team will maintain and update components
- **Community support** - Available on Langflow Discord
- **Enterprise support** - Direct support for enterprise users
- **Documentation** - Comprehensive guides and examples available

## 🎯 Future Enhancements

- Additional data extraction schemas
- Advanced proxy configuration options
- Real-time data monitoring
- Custom webhook integrations

---

**Ready to supercharge your AI workflows with enterprise web data!** 🌟

For questions or support, reach out to the Bright Data team or join the discussion in the Langflow community.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Introduced three new components for enhanced web search, structured data extraction, and web scraping using Bright Data services.
  * Added support for Google, Bing, and Yandex search engines, AI-powered dataset detection, and bot detection bypass in scraping.
  * Integrated a new Bright Data icon and color scheme in the user interface.
  * Added Bright Data to the sidebar bundles for easier access.

* **Style**
  * Updated UI icon and color mappings to visually distinguish Bright Data components.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Idanvilenski committed:
feat: v1.0 - added a Bright Data bundle with 3 components

Idanvilenski committed:
Tool Update

Idanvilenski committed:
Update Brightdata components

Idanvilenski committed:
feat: Add Bright Data integration bundle

- Add BrightDataWebScraperComponent for web scraping
- Add BrightDataSearchEngineComponent for search results
- Add BrightDataStructuredDataEnhancedComponent for data extraction
- Include custom Bright Data icons and branding
- Support tool mode for agent integration
- Fix all linting issues and type annotations
- Update component registry for proper imports

Idanvilenski committed:
Merge remote-tracking branch 'brightdata-fork/main' into main

Idanvilenski committed:
fix: Address code review feedback

- Fix component name consistency between frontend and backend
- Resolve duplicate dataset ID issue
- Clean up unused imports
- Fix formatting issues

Idanvilenski committed:
feat: fixed all the errors to authorize the pull request

Idanvilenski commented:
Hi @mendonk, I saw that you had previously reviewed some of the PRs, can you please take a look? 
Thanks,
Idan

mendonk commented:
@Idanvilenski Hi there! I am just the reviewer for documentation PRs, but I'll see if I can get someone from eng team to review your PR. Thanks for your contribution!

mendonk commented:
@Idanvilenski Can you please add tests to your submission? A member of the team can review it once tests are added. Thank you.

Idanvilenski committed:
Add comprehensive test suite for Bright Data components

- Add unit tests for all three components with mocked API calls
- Add integration tests for component structure validation
- Add component initialization and import tests
- Add manual testing documentation
- Achieve high test coverage without requiring real API keys

Idanvilenski committed:
Merge upstream changes and resolve conflicts
